File: metadata/src/application/dtos/graph_schema_dto.rs
use crate::domain::models::GraphSchemaModel;
use bric_a_brac_dtos::GraphSchemaDto;

impl From<GraphSchemaModel> for GraphSchemaDto {
    fn from(schema: GraphSchemaModel) -> Self {
        Self {
            nodes: schema.nodes.into_iter().map(From::from).collect(),
            edges: schema.edges.into_iter().map(From::from).collect(),
        }
    }
}

File: web-ui/lib/api/dtos/graph-schema-dto.ts
import * as v from "valibot";
import { EdgeSchemaDto } from "./edge-schema-dto";
import { NodeSchemaDto } from "./node-schema-dto";

export const GraphSchemaDto = v.object({
  nodes: v.array(NodeSchemaDto),
  edges: v.array(EdgeSchemaDto)
});
